### PR TITLE
Dont change old ids in format

### PR DESCRIPTION
**Pull request by DinaMeylakh**
## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://github.com/demisto/etc/issues/46901)

### CODE REVIEW

**Comment by DinaMeylakh** [comment]
Failing unit-test for TDD: [url](https://app.circleci.com/pipelines/github/demisto/demisto-sdk/15955/workflows/25349606-29b1-46da-80d8-5f599475e2b9/jobs/95952) with [commit](https://github.com/demisto/demisto-sdk/pull/2040/commits/65d56c68f79131aa21bc99252113e52c2006e287)

File: demisto_sdk/commands/format/update_generic_json.py [hunk 1]
```diff
@@ -93,18 +93,24 @@ def remove_null_fields(self):
     def update_id(self, field='name') -> None:
         """Updates the id to be the same as the provided field ."""
         updated_integration_id_dict = {}
+        should_update_id = True
+        if self.old_file:
+            current_id = self.data.get('id')
+            old_id = self.old_file.get('id')
+            should_update_id = (current_id != old_id)
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/format/update_generic_json.py:100]: In the case where the current id is different from the existing id (meaning the developer has edited the id), you would not want to overwrite it as you do in lines 109-110.

File: demisto_sdk/commands/format/tests/test_formatting_json_test.py [hunk 2]
```diff
@@ -1292,3 +1292,46 @@ def test_json_run_format_old_classifier(self, mocker, pack):
...
+    json_object = BaseUpdateJSON(input=json_incident_type.path)
+    json_object.data['name'] = "name"
+    json_object.data['id'] = "not_name"
+    json_object.old_file = json_object.data.copy()
+    json_object.update_id()
+    assert json_object.data['id'] == "not_name"
+    assert json_object.data['name'] == "name"
+
+
+def test_updating_id_in_old_json_file(repo):
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/format/tests/test_formatting_json_test.py:1318]: Not sure this is a valid use case, if a user has changed the ID I am not sure we would want to change it for him.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # Changelog
 
+* Fixed an issue in the **format** command where the `id` field was updated in old json files.
```
Comment (ShahafBenYakir) [inline CHANGELOG.md:3]:
```suggestion
* Fixed an issue in the **format** command where the `id` field was overwritten for existing JSON files.
```

**Review by ShahafBenYakir — CHANGES_REQUESTED** [review]

File: demisto_sdk/commands/format/update_generic_json.py (hunk 1 above)
Comment (DinaMeylakh) [reply demisto_sdk/commands/format/update_generic_json.py:100]: lets discuss

File: demisto_sdk/commands/format/tests/test_formatting_json_test.py [hunk 4]
```diff
@@ -1292,3 +1292,46 @@ def test_json_run_format_old_classifier(self, mocker, pack):
...
+    """
+    Given
+        - An old json file with non matching name and id.
+        - New id modification.
+    When
+        - Run format on file.
+    Then
+        - Ensure that name id was not updated.
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/format/tests/test_formatting_json_test.py:1326]:
```suggestion
        - Ensure that the id was not updated.
```

File: demisto_sdk/commands/format/tests/test_formatting_json_test.py (hunk 2 above)
Comment (ShahafBenYakir) [reply demisto_sdk/commands/format/tests/test_formatting_json_test.py:1318]: This should be negative like the first one no?

File: demisto_sdk/commands/format/update_generic_json.py [hunk 5]
```diff
@@ -93,18 +93,25 @@ def remove_null_fields(self):
...
-
-        if updated_integration_id_dict:
-            self.updated_ids.update(updated_integration_id_dict)
+        if self.old_file:
+            current_id = self.data.get('id')
+            old_id = self.old_file.get('id')
+            if current_id != old_id:
+                click.secho(
+                    f'The modified JSON file corresponding to the path: {self.relative_content_path} ID does not match the ID in remote file.'
+                    f' Changing the ID from {current_id} back to {old_id}.', fg="yellow")
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/format/update_generic_json.py:102]:
```suggestion
                    f'The modified JSON file corresponding to the path: {self.relative_content_path} contains an ID which does not match the ID in remote file.'
                    f' Changing the ID from {current_id} back to {old_id}.', fg="yellow")
```

**Review by ShahafBenYakir — APPROVED** [review]
Small comments

**Pull request merged**